src/providers/hana/CMakeLists.txt

```
@@ -81,8 +81,20 @@ add_library(provider_hana_a STATIC ${HANA_SRCS} ${HANA_HDRS})
 target_compile_features(provider_hana PRIVATE cxx_std_17)
 target_compile_features(provider_hana_a PRIVATE cxx_std_17)
 
-target_compile_definitions(provider_hana PUBLIC ODBC_STATIC)
-target_compile_definitions(provider_hana_a PUBLIC ODBC_STATIC)
+target_compile_definitions(provider_hana
+    PUBLIC
+        ODBC_STATIC
+        "NS_ODBC=qgs::odbc"
+        "NS_ODBC_START=namespace qgs { namespace odbc {"
+        "NS_ODBC_END=} }"
+)
+target_compile_definitions(provider_hana_a
+    PUBLIC
+        ODBC_STATIC
+        "NS_ODBC=qgs::odbc"
+        "NS_ODBC_START=namespace qgs { namespace odbc {"
+        "NS_ODBC_END=} }"
+)
```

I guess the following should automatically retrieve the definitions from the odbccpp_static target, and also switch them to PRIVATE as they shouldn't be needed by users of providers_hana / providers_hana_a

```suggestion
target_compile_definitions(provider_hana PRIVATE $<TARGET_PROPERTY:odbccpp_static,INTERFACE_COMPILE_DEFINITIONS>)
target_compile_definitions(provider_hana_a PRIVATE $<TARGET_PROPERTY:odbccpp_static,INTERFACE_COMPILE_DEFINITIONS>)
```

---

Thanks. Done in 4aa326c12bc3a549d9edd42edfb8eeb447ebbc7c.